src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```
@@ -0,0 +1,41 @@
+package mate.academy.security;
+
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final int MINIMUM_PASSWORD_LENGTH = 8;
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        User user = userService.findByEmail(email).orElseThrow(() ->
+                new AuthenticationException("Something wrong with login or password"));
+        String hashedPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!user.getPassword().equals(hashedPassword)) {
+            throw new AuthenticationException("Something wrong with login or password");
+        }
+        return user;

```

Can we throw this exception only once?

---

> Can we throw this exception only once?

definitely